### DISCUSSION TITLE
missing go.sum entry

### DISCUSSION THREAD

**Discussion by robotLJW** [golangci/golangci-lint#1958] [Q&A] [answered]
In a Github CI build, I hit the issue of golangci-lint. The project relies on a third-party library that does not have a go. sum file. 
Please help take a look at it. Any directions would be appreciated. Thanks.

More detailed log is at https://github.com/robotLJW/servicecomb-service-center/runs/2518134210

> Running [/home/runner/golangci-lint-1.39.0-linux-amd64/golangci-lint run --out-format=github-actions --timeout=5m --skip-dirs='test,.*/controller/(v3|v4)$,.*/bootstrap$,server/broker,examples,frontend,scctl,integration' --enable golint,gocyclo,goimports --skip-files=.*_test.go$] in [] ...
  level=error msg="Running error: context loading failed: failed to load packages: failed to load with go/packages: err: exit status 1: stderr: go: github.com/jinzhu/copier@v0.3.0: missing go.sum entry; to add it:\n\tgo mod download github.com/jinzhu/copier\n"

Third party Library: https://github.com/jinzhu/copier/blob/master/go.mod.

Do I need to delete the go.mod file or add the go.sum file.

**Comment by ldez** [comment, marked as answer]
Hello,

Your problem is not because `copier` doesn't have a `go.sum` file but because your project doesn't have a `go.sum` file

You have to add the `go.sum` file by executing the following command: `go mod tidy`

You also have to remove this line: https://github.com/apache/servicecomb-service-center/blob/9d50638496e37fa3e5bfac388e382f932430b10c/.gitignore#L42

https://github.com/jinzhu/copier doesn't have to add a `go.sum` file because there are no dependencies.

> Ensure your go.sum file is committed along with your go.mod file. 

https://github.com/golang/go/wiki/Modules#releasing-modules-all-versions

> Both go.mod and go.sum should be checked into version control.

https://blog.golang.org/using-go-modules#TOC_3.

**Reply by robotLJW** [reply]
Thanks.
